packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -240,6 +256,7 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
     if (
       // If this version's migration has already been completed we can proceed
       Either.isRight(aliasesRes) &&
+      // TODO check that this behaves correctly when skipping reindexing
```

Comment:
Do we do this TODO now 😄 ?

Reply:
Checked! I was afraid that the `WAIT_FOR_MIGRATION_COMPLETION` would not work if we didn't generate a new index with the current version, e.g. `.kibana_8.8.0_001`. But the fact is that the check is performed against the aliases, and these are correctly updated to point to the previous version index (if we skip reindex).